src/main/java/com/nvidia/spark/rapids/jni/GpuTimeZoneDB.java

```diff
@@ -230,6 +269,11 @@ private void doLoadData() {
                   );
                 }
               });
+              ZoneOffsetTransition last = transitions.get(transitions.size() - 1);
```

Comment:
Seems it's OK to add a max item in the end.
Not sure if add a max item fixed a bug.

If the searched item is the last item, then `idx-1` refer to the wrong index.
Refer to the following example:
Search `8` get end(), then `idx -1` is not want we want.

[upper_bound](https://thrust.github.io/doc/group__binary__search_gac00fd24a457a398e91c3d3822e02b95a.html#gac00fd24a457a398e91c3d3822e02b95a):
```cpp
thrust::device_vector<int> input(5);
input[0] = 0;
input[1] = 2;
input[2] = 5;
input[3] = 7;
input[4] = 8;
thrust::upper_bound(thrust::device, input.begin(), input.end(), 0); // returns input.begin() + 1
thrust::upper_bound(thrust::device, input.begin(), input.end(), 1); // returns input.begin() + 1
thrust::upper_bound(thrust::device, input.begin(), input.end(), 2); // returns input.begin() + 2
thrust::upper_bound(thrust::device, input.begin(), input.end(), 3); // returns input.begin() + 2
thrust::upper_bound(thrust::device, input.begin(), input.end(), 8); // returns input.end()
thrust::upper_bound(thrust::device, input.begin(), input.end(), 9); // returns input.end()
```

Another option: use `lower_bound` instead.
then:  `it != end && *it = searched_item` to find an item.
Then we can remove the min and max items.

I propose to remove the min/max items and use `lower_bound`. Should be a follow-up.

Reply:
Be careful when there is no item in the vector,  `end - 1` will point to a wrong address.

Reply:
> Be careful when there is no item in the vector, `end - 1` will point to a wrong address.

Yes, since there is a start point and an end point, I think this case will not happen.